### ISSUE TITLE
Installer loops on checking package validity when package is spelled incorrectly

### ISSUE THREAD

**Issue by mhussaincov93** [archlinux/archinstall#951]
stepps to repro:
1.
when you go to add extra packages that you would like to be installed, add a spelling error like
pccmanfm
what should happen:
you should be returned to the box where you typed in your list of extra packages with the box being emptied so that you may try again.
before being returned to the prya box, you should be told that x package was not found.
in this case pccmanfm.
what happens:
archinstall enteres a loop of checking if the packages that you've entered are present.
trying this from latest iso.
Majid

**Comment by svartkanin** [comment]
I've actually encountered a different error as well when adding additional packages and just entering rubbish

```
Write additional packages to install (space separated, leave blank to skip): asdf
Verifying that additional packages exist (this might take a few seconds)
Traceback (most recent call last):
  File "/usr/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/archinstall/archinstall/__main__.py", line 15, in <module>
    archinstall.run_as_a_module()
  File "/archinstall/./archinstall/__init__.py", line 267, in run_as_a_module
    script.execute()
  File "/archinstall/archinstall/lib/profiles.py", line 194, in execute
    self.spec.loader.exec_module(sys.modules[self.namespace])
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/archinstall/archinstall/examples/guided.py", line 256, in <module>
    ask_user_questions()
  File "/archinstall/archinstall/examples/guided.py", line 90, in ask_user_questions
    global_menu.run()
  File "/archinstall/archinstall/lib/menu/selection_menu.py", line 261, in run
    if not self._process_selection(selection):
  File "/archinstall/archinstall/lib/menu/selection_menu.py", line 278, in _process_selection
    return self.exec_option(selector_name,selector)
  File "/archinstall/archinstall/lib/menu/selection_menu.py", line 297, in exec_option
    result = selector.func()
  File "/archinstall/archinstall/lib/menu/selection_menu.py", line 486, in <lambda>
    lambda: ask_additional_packages_to_install(storage['arguments'].get('packages', None)),
  File "/archinstall/archinstall/lib/user_interaction.py", line 408, in ask_additional_packages_to_install
    validate_package_list(packages)
  File "/archinstall/archinstall/lib/packages/packages.py", line 89, in validate_package_list
    invalid_packages = [
  File "/archinstall/archinstall/lib/packages/packages.py", line 92, in <listcomp>
    if not find_package(package)['results'] and not find_group(package)
TypeError: 'PackageSearchResult' object is not subscriptable
```

**Comment by svartkanin** [comment]
I'll have a look at both of them, maybe they originate at the same point

**Comment by Torxed** [comment]
PackageSearchResult is a new thing, which should help fix this issue tho. But that error should be fixed too

**Comment by svartkanin** [comment]
@dylanmtaylor the fix got merged with #959 we just haven't closed this yet
